src/app/components/hris/employees/view-employee/view-employee.component.ts

```diff
@@ -425,4 +513,8 @@ export class ViewEmployeeComponent {
     const result = capitalizedWords.join(' ');
     return result;
   }
+
+  sortTerminatedNames(a: string, b: string, isAsc: boolean) {
```

Comment:
@MathildaShongwe15 

Perhaps it may be better changing to SelectedEmployeeName and ComparedEmployeeName

Reply:
@bseptember  I did update the naming of the variables to more descriptive. Please check updated changes in code.

@LlewellynAnthony 

This function takes in two names at a time.  (EmployeeName1 and EmployeeName2 from the array of names in that column)

**isAsc** - is a boolean that checks for ascending which would be true or descending which would be false.

The function compares the two strings (e.g **EmployeeName1** = Anna and **EmployeeName2** = Brandy) in this case: 
if statement will compare the two names:  EmployeeName1 comes before EmployeeName2 alphabetically. In that case , -1 will be assigned and vice versa.  

The "isAsc" value comes from this  **const isAsc = sort.direction === 'asc';** which sets the sort direction beforehand.

This is will return true and that will be return 1 otherwise -1 if set to descending.

These two values will multiplied and **(-1 * 1) = -1**. This will return -1 on the function below.

 **switch (sort.active) {
        case 'terminatedNames': return this.sortTerminatedNames(employeeName1.name, employeeName2.name, isAsc);
        default: return 0;
      }**

so if there is no values to compare on this column 'terminatedNames' it will default to 0.

This function  **this.terminatedDataSource.data = this.terminatedDataSource.data.sort((employeeName1: any, employeeName2: any) => {}** expects a return  integer number in order for it to sort the names.